### ISSUE TITLE
Make "missing parameter value" message friendlier

### ISSUE THREAD

**Issue by arikfr** [getredash/redash#2721]
Currently we piggy back the query execution error mechanism to notify the user that they need to enter a value for a parameter when a value is missing.

On a query screen:

![image](https://user-images.githubusercontent.com/71468/43687881-737da8fc-98e6-11e8-8c94-2aed32dc2680.png)

On a dashboard:

![image](https://user-images.githubusercontent.com/71468/43687884-7c4967fa-98e6-11e8-8987-145cbd1978e5.png)

The big red message feels very intimidating and I wonder if there is a better way to convey this message?

**Comment by kocsmy** [comment]
How about changing the design to warning or info so it'll be less intimidating? We can also refine the text to be friendlier: Please set a value for example parameter.

**Comment by kocsmy** [comment]
We should add the error message next to the input fields.

**Comment by arikfr** [comment]
We shouldn't treat this as an error message. It's just a state of the UI. One possible solution is to mark the parameters which miss value with some indication.

**Comment by kocsmy** [comment]
I meant like an input validation error, Bootstrap has this supported by default:
<img width="374" alt="screenshot 2018-08-27 17 59 38" src="https://user-images.githubusercontent.com/2378022/44670497-19e7d080-aa23-11e8-93a3-7f8f4c7534fe.png">

**Comment by thoechsmann** [comment]
Would also be good to allow for empty parameters. Sometimes that is intended and I have to make a workaround like enter a text like 'none' when string parameter should be empty.

**Comment by kocsmy** [comment]
@arikfr I'm wondering if we should move those parameters inputs to Ant. Their error message is similar but placed below the form (which makes more sense to me). Furthermore, the date and time pickers are already in Ant.

![image](https://user-images.githubusercontent.com/2378022/48271892-a78d6680-e43d-11e8-819b-9ce62fb284ef.png)

**Comment by arikfr** [comment]
I think that @kravets-levko's implementation of dashboard parameters already switches everything to React, so using Ant won't be a problem.

But --

1. I'm not sure we should show any message beyond marking the field.
2. I'm looking for a more holistic definition of the behavior here when trying to run a query (or refresh a dashboard) and some parameter is missing value.

**Comment by kocsmy** [comment]
1. I think we should show error message. Traditionally, the UX of error message should 1. show where the error is, 2. tell what the error is, 3. recommend a solution. For example: Missing parameter, please fill. (or something similar)

2. Can you elaborate on this and what's the problem with the current implementation? The interactions feel quite natural for me and besides moving the error message to its place (next to the field), I can't see serious issues here but I might be missing something.

**Comment by arikfr** [comment]
> Can you elaborate on this and what's the problem with the current implementation?

It looks ugly and intimidating.

I updated the issue description with the proposed solution. See if you have any comments.

**Comment by kocsmy** [comment]
Yes, this is better this way. Do you think we should just highlight the input without placing any error message below (like in my proposed design). I think it can be enough just to highlight the field but not 100% convinced.

**Comment by arikfr** [comment]
I think for now the highlight is enough. In most cases the issue is missing value, and this is obvious from the empty highlighted field.

**Comment by ranbena** [comment]
I think that best approach is to disable the "Execute" button when not all execute conditions are fulfilled, alongside a clear indication of the disable reason.

(Working on demo)

**Comment by arikfr** [comment]
Looking forward :)

**Comment by ranbena** [comment]
(ignore the wording and style)
Disallow execution if any param empty.
1. Toggle off execute button
2. Show reason tooltip on button
3. Indicate error on empty parameters. (Error message is optional and might be better without)

<img width="500" alt="screen shot 2019-01-15 at 13 26 41" src="https://user-images.githubusercontent.com/486954/51177831-73b14000-18c9-11e9-8a47-8bb3072e6dad.png">

**Comment by arikfr** [comment]
@ranbena makes sense. I also think we can skip the error message, and just use the error indication.

It does mean we will need to duplicate the logic of finding missing values between backend and frontend.

cc: @rauchy

**Comment by ranbena** [comment]
> When a required parameter value is missing

@arikfr are all params required?

**Comment by arikfr** [comment]
@ranbena yes :-) but it will change eventually. 

Some additional input:

1. There might be other issues with a parameter, like wrong value.
2. Based on the above and to avoid duplicate logic between backend/frontend, how about we change the backend response to have a dictionary of "parameter name" => "error message" that we will use here?

**Comment by ranbena** [comment]
> 1. There might be other issues with a parameter, like wrong value.
> 2. Based on the above and to avoid duplicate logic between backend/frontend, how about we change the backend response to have a dictionary of "parameter name" => "error message" that we will use here?

Sounds good.
Yay my first backend task!

**Comment by jekeam** [comment]
Why are all parameters required, I don't like this approach. I would like to create an optional parameter, but this is not possible?
